### PR TITLE
Fix publishing setup

### PR DESCRIPTION
**Pull request by margaritajankovska**
Followed halley and prince of versions config setups and tweaked bits and pieces to make it work. 

Also added debug deploy task that deploys to local maven

### CODE REVIEW

**Review by Aksi0m — APPROVED** [review]
Looks good. Thanks!

File: tasks.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 task javadocs(type: Javadoc) {
     source = android.sourceSets.main.java.source
     classpath += project.files(android.getBootClasspath().join(File.pathSeparator))
+    excludes = ['**/*.kt']
```
Comment (antunflas) [inline tasks.gradle:4]: Why do we exclude kt files from the docs?

File: deploy.gradle [hunk 2]
```diff
@@ -12,4 +12,14 @@ task deployGoldfingerRx(dependsOn: [
 ]) {
     group = "Deploy"
     description = "Deploy module to repositories"
+}
+
+task deployDebug(dependsOn: [
```
Comment (antunflas) [inline deploy.gradle:17]: Could you name this deployLocal because debug is misleading here?

File: tasks.gradle (hunk 1 above)
Comment (spasepremcheski) [reply tasks.gradle]: Margarita was getting an error trying to deploy locally without this, but I just checked and I successfully deployed locally so I removed this line

File: deploy.gradle (hunk 2 above)
Comment (spasepremcheski) [reply deploy.gradle]: Yes, done

File: core/publish.gradle [hunk 3]
```diff
@@ -26,16 +26,18 @@ afterEvaluate {
...
         publications {
             release(MavenPublication) {
+                from components.release
+
                 groupId = releaseConfig.groupId
                 version = versions.goldfinger
 
                 artifactId = 'goldfinger'
 
-                artifact bundleReleaseAar
```
Comment (KCeh) [inline core/publish.gradle:34]: Is there a need to remove this and add
`packaging = "aar"` in pom block?
Both modules are android libraries (not java libraries) 

If nobody knows I can recheck it

Comment (spasepremcheski) [reply core/publish.gradle:34]: Could you please recheck what is the core difference between those two? 
I found that `packaging = "aar"` is used as metadata to tell the consumers that this is aar, but `artifact bundleReleaseAar` specifies the file to publish. I think that we might want to use a combination of both.

Comment (KCeh) [reply core/publish.gradle:34]: Yea, sure I will look into it when I get time

Comment (KCeh) [reply core/publish.gradle:34]: When using `artifact bundleReleaseAar` I was actually getting error.
But removing it, and removing `packaging = "aar"` resulted in same pom file.
So option `packaging = "aar"` doesn't seem to do anything 🤷‍♂️ 
But we can leave it, no need to change anything now

**Review by spasepremcheski — APPROVED** [review]

**Pull request merged**